analyzer/tests/functional/analyze/test_analyze.py

```diff
@@ -84,7 +84,9 @@ def __analyze_incremental(self, content_, build_json, reports_dir,
         print(err)
 
         errcode = process.returncode
-        self.assertEqual(errcode, 0)
+        # This function checks incremental analysis. Sometimes the analysis
+        # succeeds, sometimes it fails.
+        self.assertIn(errcode, [0, 3])
```

Comment:
why does this fail sometimes?

Reply:
I rephrased it: during incremental analysis we test what happens when one analysis is successful and another fails. This is a common function which runs analysis in both cases.